swo_aws_extension/flows/steps/setup_context.py

```diff
@@ -44,7 +49,21 @@ def setup_aws(self, context: InitialAWSContext):
             f"successfully"
         )
 
+    def init_template(self, client: MPTClient, context: InitialAWSContext):
```

Comment:
As for me, the SetupContext step now contains quite a huge amount of logic. It would be great to have a separate step to setup processing template. It could be really simple.

BTW, in that case you don't need TemplateNameManager. Since you are going to have separate step for Processing where you can have these logic

For CompleteSteps you already have different steps for Change/Purchase orders, that can have logic to choose proper template name.

Please write it down, due to timings I wouldn't like to block the PR